src/util.rs

```diff
@@ -281,9 +281,16 @@ pub mod utf8 {
 	use libc::c_char;
 
 	pub fn into<'a>(data: *const c_char, length: u32) -> Vec<&'a str> {
-		unsafe {
```

Comment:
I've pushed another commit that's a little less noisy and hopefully also clearer. Please can you take another look?